src/rpc/service_eth.go

```diff
@@ -35,20 +37,49 @@ func ComputeIdentity(prefix []byte, sender common.Address) *shcrypto.EpochID {
 	return shcrypto.ComputeEpochID(identitypreimage.IdentityPreimage(imageBytes).Bytes())
 }
 
+type EthServiceInterface interface {
+	NewBlock(ctx context.Context, blockNumber uint64)
+	SendRawTransaction(ctx context.Context, s string) (*common.Hash, error)
+}
+
 type EthService struct {
 	processor Processor
-	processedTransactions map[common.Hash]bool
+	cache     *cache.Cache
 }
 
 func (s *EthService) InjectProcessor(p Processor) {
 	s.processor = p
-	s.processedTransactions = make(map[common.Hash]bool)
+	s.cache = cache.NewCache(10) // todo make delay factor configurable
 }
 
 func (s *EthService) Name() string {
 	return "eth"
 }
 
+func (s *EthService) NewBlock(ctx context.Context, blockNumber uint64) {
+	Logger.Info().Msg(fmt.Sprintf("Received blockNumber: %d", blockNumber))
+	s.cache.Lock()
+	defer s.cache.Unlock()
+	for key, info := range s.cache.Data {
+		if info.SendingBlock == blockNumber { // todo reorg issue? <=
+			if info.Tx == nil {
+				fmt.Printf("Info is null. Deleting entry.")
+				delete(s.cache.Data, key)
+			} else {
+				fmt.Printf("Sending transaction %s to the sequencer from block listener\n", info.Tx.Hash().Hex())
+				transaction, err := s.SendRawTransaction(ctx, info.Tx.Hash().Hex())
```

Comment:
I believe the `Hash()` method will only create the tx hash and not the actual transaction bytes which need to be passed to `SendRawTransaction`